src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LevelCrossingOnRailwayCheck.java

```diff
@@ -0,0 +1,369 @@
+    }
+
+    /**
+     * Create a Return Flags for level_crossing objects.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that contains flagged issue details
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        /*-
+         * The following invalid situations are to be flagged:
+         *  1) object is node and
+         *     a) is marked as a level crossing but does in not an intersection of highway and railway
+         *     b) is not tagged as a level crossing and is an intersection of highway and railway.
+         *  2) object is not a node or point and is tagged with railway=level_crossing.
+         *  3) object is railway and intersects a highway on the same layer but there is no node.
+         */
+
+        final Optional<CheckFlag> flagIncorrectlyTagged = this.flagIncorrectlyTagged(object);
+        if (!flagIncorrectlyTagged.isEmpty())
+        {
+            return flagIncorrectlyTagged;
+        }
+        final Optional<CheckFlag> flagNonNodeTagged = this.flagNonNodeTagged(object);
+        if (!flagNonNodeTagged.isEmpty())
+        {
+            return flagNonNodeTagged;
+        }
+        final Optional<CheckFlag> flagInvalidIntersections = this.flagInvalidIntersections(object);
+        if (!flagInvalidIntersections.isEmpty())
+        {
+            return flagInvalidIntersections;
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Flag nodes incorrectly tagged with level_crossing or missing level_crossing tag.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that contains flagged issue details
+     */
+    private Optional<CheckFlag> flagIncorrectlyTagged(final AtlasObject object)
+    {
+        if (object instanceof Node)
+        {
+            final Node node = (Node) object;
+
+            final int issueType = this.isValidLevelCrossingNode(node);
+            if (Validators.isOfType(node, RailwayTag.class, RailwayTag.LEVEL_CROSSING)
+                    && issueType > 0)
+            {
+                // This is a node that is tagged with railway=level_crossing and is not a
+                // railway/highway intersection
+                return Optional.of(this
+                        .createFlag(object,
+                                this.getLocalizedInstruction(issueType, object.getOsmIdentifier()))
+                        .addFixSuggestion(FeatureChange.add(
+                                (AtlasEntity) ((CompleteEntity) CompleteEntity
+                                        .from((AtlasEntity) object)).withRemovedTag(RailwayTag.KEY),
```

Comment:
just a thought. could the railway tag just have the wrong value, but the feature is still a legitimate "railway"? in which case we may want to not send up a fix suggestion

Reply:
In this check the feature is a node, so it is not a rail way. There are valid "railway" tags for nodes, but this specific check is determining that the "level_crossing" value is not correct. I guess there could be some other valid key value for this node, but I'm not sure how to determine that. Are you saying that you don't want a fix suggestion associated with this flagged issue?

Reply:
Yeah after thinking about this for a bit we could still send up the fix suggestion. My only concern was that the editor might want to change the Railway tag value to something more appropriate if applicable, instead of deleting the tag,  but they have the ability to do that by just not applying the fix suggestion